src/TLuaInterpreter.cpp

```diff
@@ -15198,55 +15215,87 @@ void TLuaInterpreter::initIndenterGlobals()
 {
+#if defined(Q_OS_WIN32)
+    loadUtf8Filenames();
+#endif
+
     setupLanguageData();
 
+    QStringList pathsToTry = {
 #if defined(Q_OS_MACOS)
-    // Load relatively to MacOS inside Resources when we're in a .app bundle,
-    // as mudlet-lua always gets copied in by the build script into the bundle
-    QString path = QCoreApplication::applicationDirPath() + "/../Resources/mudlet-lua/lua/LuaGlobal.lua";
-#else
-    // Additional "../src/" allows location of lua code when object code is in a
-    // directory alongside src directory as occurs using Qt Creator "Shadow Builds"
-    QString path = "../src/mudlet-lua/lua/LuaGlobal.lua";
+        QCoreApplication::applicationDirPath() + "/../Resources/mudlet-lua/lua/LuaGlobal.lua",
 #endif
+        QCoreApplication::applicationDirPath() + "/mudlet-lua/lua/LuaGlobal.lua",
+        "../src/mudlet-lua/lua/LuaGlobal.lua",
+        "mudlet-lua/lua/LuaGlobal.lua",
+        LUA_DEFAULT_PATH "/LuaGlobal.lua"
+    };
+    QStringList failedMessages{};
 
-    int error = luaL_dofile(pGlobalLua, path.toUtf8().constData());
-    if (error != 0) {
-        // For the installer we do not go down a level to search for this. So
-        // we check again for the user case of a windows install.
 
-        // overload previous behaviour to check by absolute path as well
-        // TODO this sould be cleaned up and refactored to just use an array and a for loop
-        path = QCoreApplication::applicationDirPath() + "/mudlet-lua/lua/LuaGlobal.lua";
+    int error;
+    for (const auto& path : qAsConst(pathsToTry)) {
         if (!QFileInfo::exists(path)) {
-            path = "mudlet-lua/lua/LuaGlobal.lua";
+            failedMessages << tr("%1 (doesn't exist)", "This file doesn't exist").arg(path);
+            continue;
+        }
+
+        // load via Qt so UTF8 paths work on Windows - Lua can't handle it
+        auto luaGlobal = readScriptFile(path);
+        if (luaGlobal.isEmpty()) {
+            failedMessages << tr("%1 (couldn't read file)", "This file could not be read for some reason (for example, no permission)").arg(path);
+            continue;
         }
-        error = luaL_dofile(pGlobalLua, path.toUtf8().constData());
+
+        error = luaL_dostring(pGlobalLua, luaGlobal.toUtf8().constData());
         if (error == 0) {
             mpHost->postMessage(tr("[  OK  ]  - Mudlet-lua API & Geyser Layout manager loaded."));
             return;
+        } else {
+            qWarning() << "TLuaInterpreter::loadGlobal() loading " << path << " failed: " << lua_tostring(pGlobalLua, -1);
+            failedMessages << QStringLiteral("%1 (%2)").arg(path, lua_tostring(pGlobalLua, -1));
         }
-    } else {
-        mpHost->postMessage(tr("[  OK  ]  - Mudlet-lua API & Geyser Layout manager loaded."));
+    }
+
+    mpHost->postMessage(tr("[ ERROR ] - Couldn't to load LuaGlobal; your Mudlet is broken! Tried these locations:\n%1").arg(failedMessages.join(QStringLiteral("\n"))));
```

Comment:
`"[ ERROR ] - Couldn't to load LuaGlobal; your Mudlet is broken! Tried these locations:\n%1"`
That "to" is surplus, how about:
`"[ ERROR ] - Couldn't find and load LuaGlobal.lua; your Mudlet is broken!\nTried these locations:\n%1"`

BTW `QStringList::join(...)` supports `QChar` for a while now so you can propbably use the more compact: `.join(QChar::LineFeed)` :grin: